### DISCUSSION TITLE
How do I move the replacement `%s//gc` "Confirm" box from center?

### DISCUSSION THREAD

**Discussion by hasecilu** [LazyVim/LazyVim#3983] [Q&A] [answered]
When I use `%s/key/replacement/gc` to make some string replacements with confirmation on current file when moving to next match the key is moved to the center and also the confirmation box is moved to the center making that a lot of times **I can not see the string** which invalidates the whole point of the command making me to do checks manually. Do you think should be considered a bug on LazyVim?

It should be moved up because when it searches for next match puts it in the enter, if next match is visible it won't move from current line so lines on bottom half are visible. Probably having it in the same position as when command prompt box (`:`) was created is good option.

![image](https://github.com/LazyVim/LazyVim/assets/53124818/03662c38-5899-4626-bf2a-1ae5698bf979)

**Comment by dpetka2001** [comment]
It's a view from `noice.nvim`. You should read the docs of the plugin if you want to change some setting. This is definitely not a bug.

**Reply by hasecilu** [reply]
My reason to say it's a bug it's because there is a high change that users that use that replacement command to have that UX nuisance that can be fixable.

**Reply by dpetka2001** [reply]
It's not a LazyVim bug. If you'd like you can file an issue on `noice.nvim`'s Github repo for bad UX experience regarding the `confirm` view that is centered. But still, if you read the docs you can change it to your liking I believe, since it should be customizable (but I've not tried that myself, I just know that you can change the settings of the views).

**Comment by hasecilu** [comment, marked as answer]
At the end it was something also with `nui.nvim`. I ended up creating this file on `plugins` dir. I'm not sure if this solution is optimal but works for me. Also I don't know if there are bad consequences on other elements, still haven't detected one.

```lua
return {
  "folke/noice.nvim",
  requires = { "MunifTanjim/nui.nvim" },
  config = function()
    require("noice").setup({
      views = {
        confirm = {
          position = {
            row = "40%", -- Set the row position (e.g., 40% from the top)
            col = "50%", -- Set the column position (e.g., 50% from the left)
          },
        },
      },
    })
  end,
}
```

As you can see in the picture now there is no overlapping with the "confirm" dialog and now I can confirm/cancel replacements with 100% certainty. I think it's a better default that current one.

![image](https://github.com/LazyVim/LazyVim/assets/53124818/a6941307-273a-4307-9790-19855a202089)

**Reply by dpetka2001** [reply]
If you think it's a better default, then just create a PR on `noice.nvim` Github repo and make your case to the maintainer.

**Reply by hasecilu** [reply]
I'm checking that, I'm reading the code and I think this is the line should be changed.

https://github.com/folke/noice.nvim/blob/04ff1defa257db569a687ec16051a27fbaeb0e2f/lua/noice/config/views.lua#L244

**Reply by folke** [reply]
That makes sense. Just pushed an update to configure it to:

```lua
    position = {
      row = 3,
      col = "50%",
    },

```

**Reply by hasecilu** [reply]
I have tested the change and works fine, with confirm dialog on top you can read the match and context around. I'll mark the snippet in case some other user wants to move it.
